### DISCUSSION TITLE
[Help] Grafana container restarting

### DISCUSSION THREAD

**Discussion by tm6907** [cmu-sei/GHOSTS#322] [Q&A] [answered]
Hi so I've cloned the repo for version 8, started up the containers using docker compose and the grafana container keeps restarting over and over. In the past I was able to fix this using chown on the g_data folder as described on the docs https://cmu-sei.github.io/GHOSTS/quickstart/.

However now I can't seem to find that folder so is there something else I should be doing to resolve this?

**Comment by sei-dupdyke** [comment, marked as answer]
Should be "_g". If you don't want that hassle, you can remove the volume (the last line of the docker-compose file).

**Reply by tm6907** [reply]
Ok updated the _g folder to user:user but that hasn't resolved the issue. What does removing the last line in the compose file do? Where is it stored if not that volume?

**Reply by sei-dupdyke** [reply]
In the container.

This is not an issue for us, I just copy it in when I setup the dashboard. YMMV.

(Sigh, Grafana is great, but can be super frustrating with this restarting issue.)

**Reply by tm6907** [reply]
Ok made some progression, got the API and a new client up and running. Think I'm almost there the client has opened firefox but browsing any sites just a blank tab at the moment. No errors from the application in debug mode or in the logs. 

I realise opening the grafana dash for the first time it is full of random data, is this expected? If so would the graphs have to be changed manually to display the live data coming in? 

Thanks, for all the help.

**Reply by sei-dupdyke** [reply]
What's in the client logs?

Grafana only shows what's in the db - did you have stuff in there from prior by chance?

**Reply by tm6907** [reply]
<img width="1399" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/5199dc6f-1c1f-440f-85ca-7c253ae08f9a">
This is what the client logs were showing before. Nothing indicating any errors but it will open firefox to a blank tab only. 

The grafana and db are from a fresh install so should of started empty yeah but since running the client it should show some activity surely. 

I've been going through each graph and copying the sql query from the version 7 dash I tried.

**Reply by tm6907** [reply]
Ok the client is browsing fine now, just took a while to start. Im back to the previous issue where nothing is recorded in the history timeline or history health. I'll open a separate issue since this one is resolved. Cheers
